dartagnan/src/main/java/com/dat3m/dartagnan/Dartagnan.java

```diff
@@ -165,33 +172,17 @@ public static void main(String[] args) throws Exception {
                     modelChecker = DataRaceSolver.run(ctx, prover, task);
                 } else {
                     // Property is either PROGRAM_SPEC, LIVENESS, or CAT_SPEC
-                    switch (o.getMethod()) {
-                        case EAGER:
-                            modelChecker = AssumeSolver.run(ctx, prover, task);
-                            break;
-                        case LAZY:
-                            modelChecker = RefinementSolver.run(ctx, prover, task);
-                            break;
-                        default:
-                            throw new InvalidConfigurationException("unsupported method " + o.getMethod());
-                    }
+                    modelChecker = switch (o.getMethod()) {
+                        case EAGER -> AssumeSolver.run(ctx, prover, task);
+                        case LAZY -> RefinementSolver.run(ctx, prover, task);
+                    };
```

Comment:
I know this cover all cases right now, but why not keep the default case just in case we even add a new method and forget to update this?

Reply:
By complain you mean "raise a warning" or it won't even compile?

Reply:
It won't compile. Switches always need to be exhaustive.